pkg/wekafs/apistore.go

```diff
@@ -101,30 +104,34 @@ func (api *ApiStore) fromSecrets(ctx context.Context, secrets map[string]string,
 
 	kmsVaultNamespaceForFilesystemEncryption, ok := secrets["kmsVaultNamespaceForFilesystemEncryption"]
 	if ok {
-		preexistingVaultCreds.Namespace = strings.TrimSpace(strings.TrimSuffix(kmsVaultNamespaceForFilesystemEncryption, "\n"))
+		preexistingVaultCreds.Namespace = cleanSecretValue(kmsVaultNamespaceForFilesystemEncryption)
 	}
 
 	kmsVaultKeyIdentifierForFilesystemEncryption, ok := secrets["kmsVaultKeyIdentifierForFilesystemEncryption"]
 	if ok {
-		preexistingVaultCreds.KeyIdentifier = strings.TrimSpace(strings.TrimSuffix(kmsVaultKeyIdentifierForFilesystemEncryption, "\n"))
+		preexistingVaultCreds.KeyIdentifier = cleanSecretValue(kmsVaultKeyIdentifierForFilesystemEncryption)
 	}
 
 	kmsVaultRoleIdForFilesystemEncryption, ok := secrets["kmsVaultRoleIdForFilesystemEncryption"]
 	if ok {
-		preexistingVaultCreds.RoleId = strings.TrimSpace(strings.TrimSuffix(kmsVaultRoleIdForFilesystemEncryption, "\n"))
+		preexistingVaultCreds.RoleId = cleanSecretValue(kmsVaultRoleIdForFilesystemEncryption)
 	}
 
 	kmsVaultSecretIdForFilesystemEncryption, ok := secrets["kmsVaultSecretIdForFilesystemEncryption"]
 	if ok {
-		preexistingVaultCreds.SecretId = strings.TrimSpace(strings.TrimSuffix(kmsVaultSecretIdForFilesystemEncryption, "\n"))
+		preexistingVaultCreds.SecretId = cleanSecretValue(kmsVaultSecretIdForFilesystemEncryption)
+	}
+	organization, ok := secrets["organization"]
+	if !ok {
+		return nil, fmt.Errorf("no organization found in secret")
 	}
 
 	credentials := apiclient.Credentials{
-		Username:            strings.TrimSpace(strings.TrimSuffix(secrets["username"], "\n")),
+		Username:            cleanSecretValue(secrets["username"]),
 		Password:            strings.TrimSuffix(secrets["password"], "\n"),
```

Comment:
shouldn't we use here `cleanSecretValue` too?
I also think we should validate all keys we expect, not just `organization`